api/proto/info/info.proto

```diff
@@ -393,6 +394,42 @@ message ZCellularProvider {
   bool roaming = 4;
 }
 
+// Snapshot states
+enum ZSnapshotState {
+  Z_SNAPSHOT_STATE_UNSPECIFIED = 0;
+  // This state is used when a snapshot is in the process of being
+  // created or an error occurred during the first attempt to create it.
+  // (For example, the operation was delayed)
+  Z_SNAPSHOT_STATE_CREATING = 1;
+  // This state is used when the snapshot has been successfully created.
+  Z_SNAPSHOT_STATE_CREATED = 2;
+  // This state is used when the snapshot is pending deletion or
+  // the first deletion attempt was not successful. Also can also get
+  // into this state when snapshots were deleted automatically
+  // after the rollback command
+  Z_SNAPSHOT_STATE_DELETING = 3;
+  // This state is used when the snapshot has been successfully deleted.
+  Z_SNAPSHOT_STATE_DELETED = 4;
+}
+
+// ZInfoSnapshot - Information about snapshot in zfs for zvol
+message ZInfoSnapshot {
+  uint64 creation_time = 1;         // In seconds
+  string uuid = 2;                  // link a command and a response. Is the real name of the snapshot in ZFS
+  string volume_uuid = 3;           // Volume ID of the volume this snapshot belongs to.
+  string display_name = 4;          // Ex: "Tuesday" or creation time
+  uint64 used_space = 5;            // In byte
+  uint64 referenced = 6;            // In byte
+  double compressratio = 7;         // Compression ratio
+  uint64 snapshot_size = 8;         // In byte
+  bool encryption = 9;
+  bool readonly = 10;
+  uint64 logicalreferenced = 11;    // In byte
```

Comment:
Seems part of this options may be changed during usage of the system. If so, we should use metrics for them, info message is expected to reflect only states (and potential errors), not information about usage.

Reply:
Yes, that's right. Moved it to metrics